test/image/mocks/fixed_size_shapes.json

```diff
@@ -0,0 +1,255 @@
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 6,
+        "yanchor": -3,
+        "x0": -20,
+        "x1": 60,
+        "y0": 0,
+        "y1": 40,
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "path",
+        "xref": "x2",
+        "xsizemode": "pixel",
+        "yref": "y2",
+        "ysizemode": "pixel",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": -1,
+        "yanchor": 10,
+        "path": "M0,0 L30,0 L15,15 Z",
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "rect",
+        "xref": "x3",
+        "xsizemode": "pixel",
+        "yref": "y3",
+        "ysizemode": "pixel",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 2,
+        "yanchor": 2,
+        "x0": 3,
+        "x1": 53,
+        "y0": 20,
+        "y1": -20,
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "rect",
+        "xref": "paper",
+        "xsizemode": "scaled",
+        "yref": "paper",
+        "ysizemode": "pixel",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 0,
+        "yanchor": 0,
+        "x0": 0,
+        "x1": 0.3,
+        "y0": 10,
+        "y1": -10,
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "rect",
+        "xref": "x5",
+        "xsizemode": "scaled",
+        "yref": "y5",
+        "ysizemode": "pixel",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 0,
+        "yanchor": 10,
+        "x0": 0,
+        "x1": 10,
+        "y0": 0,
+        "y1": -30,
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "rect",
+        "xref": "x5",
+        "xsizemode": "pixel",
+        "yref": "y5",
+        "ysizemode": "scaled",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 3,
+        "yanchor": 3,
+        "x0": 0,
+        "x1": 30,
+        "y0": 0,
+        "y1": 10,
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "path",
+        "xref": "x6",
+        "xsizemode": "pixel",
+        "yref": "y6",
+        "ysizemode": "scaled",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 10,
+        "yanchor": 10,
+        "path": "M0,0 L30,0 L15,4 Z",
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      },
+      {
+        "type": "path",
+        "xref": "x6",
+        "xsizemode": "scaled",
+        "yref": "y6",
+        "ysizemode": "pixel",
+        "fillcolor": "rgba(96, 171, 50,0.7)",
+        "xanchor": 10,
+        "yanchor": 10,
+        "path": "M0,0 L10,0 L5,25 Z",
+        "line": {
+          "color": "rgba(96, 171, 50, 1)"
+        }
+      }
+    ]
+  },
+  "config": {
+    "editable": "true"
+  }
+}
```

Comment:
Very nice test image, and I think the interactions feel very natural as you have them. Other than adding a trailing newline to make the [syntax test happy](https://circleci.com/gh/plotly/plotly.js/8691?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) the only change I'd like to see is to use the other three axis types (log, date, category) on one axis each - no need to add more subplots, just convert a couple of the existing axes.